AI/VCAI/Goals.cpp

```diff
@@ -591,8 +591,40 @@ TGoalVec Explore::getAllPossibleSubgoals()
 				case Obj::REDWOOD_OBSERVATORY:
 				case Obj::PILLAR_OF_FIRE:
 				case Obj::CARTOGRAPHER:
-				case Obj::SUBTERRANEAN_GATE: //TODO: check ai->knownSubterraneanGates
 					objs.push_back (obj);
+					break;
+				case Obj::MONOLITH_ONE_WAY_ENTRANCE:
+				case Obj::MONOLITH_TWO_WAY:
+				case Obj::SUBTERRANEAN_GATE:
+					auto tObj = dynamic_cast<const CGTeleport *>(obj);
+					if(tObj)

```

Comment:
Is it really can be nullptr? Can object with  ID MONOLITH_ONE_WAY_ENTRANCE,... be not CGTeleport?